includes/class-wc-form-handler.php

```diff
@@ -205,12 +206,18 @@ public static function save_account_details() {
 		$user->last_name    = $account_last_name;
 
 		// Prevent emails being displayed, or leave alone.
-		$user->display_name = is_email( $current_user->display_name ) ? $user->first_name : $current_user->display_name;
-
+		$user->display_name = is_email( $current_user->display_name ) ? sprintf( __( '%1$s %2$s', 'display name', 'woocommerce' ), $user->first_name, $user->last_name ) : $account_display_name;
+		// Prevent display name change to email
+		if ( is_email( $user->display_name ) ) {
+			wc_add_notice( __( 'Display name cannot be changed to email address due to privacy concern.', 'woocommerce' ), 'error' );
+			$user->display_name = $current_user->display_name;
```

Comment:
Let's refactor this method to use `WC_Customer` class. This allows you to set each field and then save, rather than use wp_update_user.